### PR TITLE
deps: Bump `v8` to 10.7.193.13 and `proxy_wasm_cpp_host` to b0a0594

### LINKED ISSUE #23258: Remove `mprotect()/rwx`-related patch entry from `bazel/v8.patch`
A patch is introduced by https://github.com/envoyproxy/envoy/pull/23257 to fix  `MemoryAllocator::PartialFreeMemory()` on v8 `10.4.132.18` (https://github.com/envoyproxy/envoy/blob/ad800292a630c273fb548ac6d320522ffdbd2a8a/bazel/repository_locations.bzl#L1053). See https://github.com/envoyproxy/envoy/issues/23243 for more details. 

That patch is copied from https://chromium-review.googlesource.com/c/v8/v8/+/3700352. That commit on v8 upstream is included in `10.5` (can be checked from `10.5-lkgr`: https://chromium.googlesource.com/v8/v8.git/+/refs/heads/10.5-lkgr/src/heap/memory-allocator.cc#407) hence when we adopt 10.5 or higher, we should remove that patch (probably most of the patches will be removed/adapted anyway).

### PR DESCRIPTION
**Pull request by dio**
Commit Message: Update `v8` to 10.7.193.13 and `proxy_wasm_cpp_host` to https://github.com/proxy-wasm/proxy-wasm-cpp-host/commit/b0a0594e8ab4023857393fab349170fb487ec5a9.
Risk Level: Medium (but mostly wasm related)
Additional Description: In the process, we remove `com_googlesource_chromium_zlib`, since now v8 can be compiled without it and can be ignored: https://github.com/envoyproxy/envoy/issues/23258#issuecomment-1264980271.
 
Fixes #23258 #23390

Signed-off-by: Dhi Aurrahman <dio@rockybars.com>

### CODE REVIEW

**Comment by wrowe** [comment]
As this is still WIP, can we presume it targets main branch once the pending release has dropped around the 15th?

**Comment by dio** [comment]
Seems like it carries a fix for https://github.com/envoyproxy/envoy/issues/23390 too.

**Comment by dio** [comment]
@wrowe the bucket that provides the v8 snapshot is updated (thanks to @mpwarres). Now, this is ready to review. 

cc. @keith @phlax @PiotrSikora

File: bazel/v8.patch [hunk 1]
```diff
@@ -1,15 +1,6 @@
...
-# 4. Disable pointer compression (limits the maximum number of WasmVMs).
-# 5. Don't expose Wasm C API (only Wasm C++ API).
-# 6. Fix cross-compilation (https://crrev.com/c/3735165).
-# 7. Fix build errors in SIMD IndexOf/includes (https://crrev.com/c/3749192).
-# 8. Fix build on arm64.
-# 9. Fix build on older versions of Linux.
-# 10. Fix MemoryAllocator::PartialFreeMemory() which shouldn't try to change permissions of RWX pages,
-#     mainly affecting macOS on Apple silicon (https://crrev.com/c/3700352). This can be removed
-#     when we adopt 10.5 or higher (https://github.com/envoyproxy/envoy/issues/23258).
+# 1. Use already imported python dependencies and disable pointer compression (limits the maximum number of WasmVMs).
```
Comment (PiotrSikora) [inline bazel/v8.patch:1]: Nit: Those are 2 separate changes, so it's weird to combine them here.

**Review by PiotrSikora — APPROVED** [review]
LGTM

cc @mpwarres

File: bazel/v8.patch (hunk 1 above)
Comment (dio) [reply bazel/v8.patch:1]: I'll update the patch hence the separation is clear.

**Review by PiotrSikora — APPROVED** [review]
LGTM, thanks!

File: bazel/repository_locations.bzl [hunk 2]
```diff
@@ -1051,48 +1051,31 @@ REPOSITORY_LOCATIONS_SPEC = dict(
...
+        # NOTE: Update together with com_googlesource_chromium_base_trace_event_common.
+        version = "10.7.193.13",
         # Static snapshot created using https://storage.googleapis.com/envoyproxy-wee8/wee8-fetch-deps.sh.
-        sha256 = "154223731091f531a3cb1a95ab9c14225983891240220a9e232c729358badd2c",
+        sha256 = "2170df76ce5d7ecd7fb8d131370d210152f200273cba126f06d8b88fb53c9fbc",
         urls = ["https://storage.googleapis.com/envoyproxy-wee8/v8-{version}.tar.gz"],
         use_category = ["dataplane_ext"],
         extensions = ["envoy.wasm.runtime.v8"],
-        release_date = "2022-07-11",
+        release_date = "2022-12-09",
```
Comment (wrowe) [inline bazel/repository_locations.bzl:1061]: Hmmm? That would be 2 months in the future

**Review by wrowe — CHANGES_REQUESTED** [review]

File: bazel/repository_locations.bzl (hunk 2 above)
Comment (phlax) [reply bazel/repository_locations.bzl:1061]: wierd - how is that passing dep checks

Comment (phlax) [reply bazel/repository_locations.bzl:1061]: ah - i see its not a github dep

Comment (dio) [reply bazel/repository_locations.bzl]: Ah, sorry I thought I copy-pasted it right from here: https://dev.azure.com/cncf/envoy/_build/results?buildId=118312&view=logs&jobId=1c04237c-5f52-5959-5f83-5522b3556331&j=1c04237c-5f52-5959-5f83-5522b3556331&t=db3f3889-c922-566b-fa2e-4bc4948bc467.

**Comment by wrowe** [comment]
/lgtm deps 

Thanks, for dropping a dependency this deserves to land in the release. Appreciated!

File: bazel/repository_locations.bzl (hunk 2 above)
Comment (dio) [reply bazel/repository_locations.bzl]: From:

```
curl -v https://storage.googleapis.com/envoyproxy-wee8/v8-10.7.193.13.tar.gz > /dev/null
```

```
last-modified: Wed, 12 Oct 2022 20:05:56 GMT
```

**Comment by dio** [comment]
Thanks, @wrowe! Seems like we need your PR review approval as well. :)

**Review by PiotrSikora — APPROVED** [review]

**Comment by wrowe** [comment]
@PiotrSikora, @lizan this has all it's stamps. Please apply today if you feel certain it is beneficial and not disruptive to the imminent release tag.

**Review by phlax — APPROVED** [review]
lgtm, thanks @dio

**Comment by dio** [comment]
Thanks, for merging @phlax! macOS 1.24.x build for homebrew will be interesting!

**Pull request merged**